galsim/phase_psf.py

```diff
@@ -0,0 +1,845 @@
+    layers or optical phase screens.  Generally, one would assemble a PhaseScreenList object using
+    the function `Atmosphere`.  Layers can be added, removed, appended, etc. just like items can be
+    manipulated in a python list.  For example:
+
+    # Create an atmosphere with three layers.
+    > screens = galsim.PhaseScreenList([galsim.AtmosphericScreen(...),
+                                        galsim.AtmosphericScreen(...),
+                                        galsim.AtmosphericScreen(...)])
+    # Add another layer
+    > screens.append(galsim.AtmosphericScreen(...))
+    # Remove the second layer
+    > del screens[1]
+    # Switch the first and second layer.  Silly, but works...
+    > screens[0], screens[1] = screens[1], screens[0]
+
+    Note that creating new PhaseScreenLists from old PhaseScreenLists copies the wrapped phase
+    screens by reference, not value.  Thus, advancing the screens in one list will also advance the
+    screens in a copy:
+
+    > more_screens = screens[0:2]
+    > more_screens.advance()
+    > assert more_screens[0] == screens[0]
+
+    > more_screens = galsim.PhaseScreenList(screens)
+    > screens.reset()
+    > psf = screens.makePSF(exptime=exptime, ...)        # starts at t=0
+    > psf2 = more_screens.makePSF(exptime=exptime, ...)  # starts at t=exptime
+    > assert psf != psf2
+
+    Methods
+    -------
+    makePSF()          Obtain a PSF from this set of phase screens.  See PhaseScreenPSF docstring
+                       for more details.
+    advance()          Advance each phase screen in list by self.time_step.
+    advance_by()       Advance each phase screen in list by specified amount.
+    reset()            Reset each phase screen to t=0.
+    path_difference()  Compute the cumulative pathlength difference due to all screens.
+
+    @param layers  Sequence of phase screens.
+    """
+    def __init__(self, layers):
+        self._layers = layers
+        self._update_attrs()  # for now, just updating self.time_step
+
+    def __len__(self):
+        return len(self._layers)
+
+    def __getitem__(self, index):
+        import numbers
+        cls = type(self)
+        if isinstance(index, slice):
+            return cls(self._layers[index])
+        elif isinstance(index, numbers.Integral):
+            return self._layers[index]
+        else:
+            msg = "{cls.__name__} indices must be integers"
+            raise TypeError(msg.format(cls=cls))
+
+    def __setitem__(self, index, layer):
+        self._layers[index] = layer
+        self._update_attrs()
+
+    def __delitem__(self, index):
+        del self._layers[index]
+        self._update_attrs()
+
+    def append(self, layer):
+        self._layers.append(layer)
+        self._update_attrs()
+
+    def extend(self, layers):
+        self._layers.extend(layers)
+        self._update_attrs()
+
+    def __str__(self):
+        return "galsim.PhaseScreenList(%s)" % self._layers
```

Comment:
This looks bad, since `str(_layers)` uses the repr of the elements in the list.  You should use something like:
```
return "galsim.PhaseScreenList([%s])" % ",".join([str(l) for l in self._layers])
```
I think this will look a bit nicer for the str version.